src/mgr.sol

```diff
@@ -159,64 +186,80 @@ contract TinlakeManager is LibNote {
         vat.slip(ilk, address(this), int256(wad));
         vat.frob(ilk, address(this), address(this), address(this), int256(wad), 0);
+        emit Join(wad);
     }
 
-    function exit(uint256 wad) public ownerOnly note {
-        require(safe && live);
+    function exit(uint256 wad) public operatorOnly {
+        require(safe && live, "TinlakeManager/bad-state");
         require(wad <= 2 ** 255, "TinlakeManager/overflow");
         vat.frob(ilk, address(this), address(this), address(this), -int256(wad), 0);
         vat.slip(ilk, address(this), -int256(wad));
         gem.transfer(msg.sender, wad);
+        emit Exit(wad);
     }
 
     // draw & wipe call daiJoin.exit/join immediately
-    function draw(uint256 wad) public ownerOnly note {
-        require(safe && live);
+    function draw(uint256 wad) public operatorOnly {
+        require(safe && live, "TinlakeManager/bad-state");
         (, uint256 rate, , , ) = vat.ilks(ilk);
         uint256 dart = divup(mul(RAY, wad), rate);
         require(int256(dart) >= 0, "TinlakeManager/overflow");
         vat.frob(ilk, address(this), address(this), address(this), 0, int256(dart));
         daiJoin.exit(msg.sender, wad);
+        emit Draw(wad);
     }
 
-    function wipe(uint256 wad) public ownerOnly note {
-        require(safe && live);
+    function wipe(uint256 wad) public operatorOnly {
+        require(safe && live, "TinlakeManager/bad-state");
         dai.transferFrom(msg.sender, address(this), wad);
         daiJoin.join(address(this), wad);
         (,uint256 rate, , , ) = vat.ilks(ilk);
         uint256 dart = mul(RAY, wad) / rate;
         require(dart <= 2 ** 255, "TinlakeManager/overflow");
         vat.frob(ilk, address(this), address(this), address(this), 0, -int256(dart));
+        emit Wipe(wad);
     }
 
     // --- Administration ---
-    function setOwner(address newOwner) external ownerOnly note {
-        owner = newOwner;
+    function setOperator(address newOperator) external operatorOnly  {
+        operator = newOperator;
+        emit NewOperator(newOperator);
     }
 
-    function migrate(address dst) public auth note {
+    function migrate(address dst) public auth  {
         vat.hope(dst);
         dai.approve(dst, uint256(-1));
         gem.approve(dst, uint256(-1));
         live = false;
+        emit Migrate(dst);
+    }
+
+    function file(bytes32 what, address data) public auth {
+        require(live, "TinlakeManager/not-live");
+        emit File(what, data);
+        if (what == "vow") vow = data;
+        else if (what == "join") daiJoin = JoinLike(data);
+        else if (what == "end")  end = EndLike(data);
+        else revert("Vat/file-unrecognized-param");
     }
 
     // --- Liquidation ---
-    function tell() public note {
-        require(safe);
-        require(wards[msg.sender] == 1 || (msg.sender == owner && !live), "TinlakeManager/not-authorized");
+    function tell() public auth {
```

Comment:
It seems `tell` can't be called anymore by the operator if shutdown happens. Is that an intentional change?
If this change was actually wrong, couldn't we even make that this is called by anyone (if shutdown happens)?

Reply:
Yes, previously the only way to set the live flag to false was through the `migrate` call and by calling `cage` but cage requires a tell call beforehand. 

In the end the operator doesn't to have the power to trigger the `tell`.

That was the reason why @MrChico removed it.

@MrChico , correct?

Reply:
Hmm not really, `cage` can be called by anyone if shutdown happens in the MCD system. Prior to this PR, the operator could run `tell` if the `!live`.

Reply:
Yes, removing the ability for the operator to call `tell` was intentional. The operator can de-risk the position in other ways, and removing this privilege for the operator reduces some complexity.

Note that `mgr.cage()` requires `!glad` which implies that `tell` has already been called.

Reply:
Ah yes, you are right, so `cage` can't be called if governance hasn't previously called `tell`.
Tbh I need to draw a states flow chart to understand well all these cases.

Reply:
Yeah, we should probably add one to the README